addons/osfstorage/views.py

```diff
@@ -220,9 +224,42 @@ def osfstorage_get_children(file_node, **kwargs):
                 WHERE P._id = 'download:' || %s || ':' || F._id
                 LIMIT 1
             ) DOWNLOAD_COUNT ON TRUE
+            LEFT JOIN LATERAL (
+              SELECT EXISTS(
+                SELECT (1) FROM osf_fileversion
+                INNER JOIN osf_basefilenode_versions ON (osf_fileversion.id = osf_basefilenode_versions.fileversion_id)
+                INNER JOIN osf_fileversionusermetadata ON (osf_fileversion.id = osf_fileversionusermetadata.file_version_id)
+                INNER JOIN osf_osfuser ON (osf_fileversionusermetadata.user_id = osf_osfuser.id)
+                INNER JOIN osf_guid ON (osf_osfuser.id = osf_guid.object_id AND (osf_guid.content_type_id = %s))
+                WHERE (osf_basefilenode_versions.basefilenode_id = F.id AND osf_guid._id = %s)
+                LIMIT 1
+              )
+            ) SEEN_FILE ON TRUE
+            LEFT JOIN LATERAL (
+                SELECT CASE WHEN SEEN_FILE.exists
+                THEN
+                    CASE WHEN EXISTS(
+                        SELECT (1) FROM osf_osfuser
+                        INNER JOIN osf_fileversionusermetadata ON (osf_osfuser.id = osf_fileversionusermetadata.user_id)
+                        INNER JOIN osf_guid ON (
```

Comment:
This `JOIN` duplicates the one in the query above. It might be more efficient to get the user's PK once at the top of the function. That should also simplify the queries here.

```python
user_pk = OSFUser.objects.filter(guids___id=user_id).values_list('pk', flat=True).get()
with connection.cursor() ....:
```